vtex-sites/base.store
# chore: Add icons to unique svg file and remove from final bundle

First of all, I want to thank @tlgimenes for the contribution and help.
## What's the purpose of this pull request?
Remove inline SVGs icons from the final bundle and create a unique SVG file with all SVGs as symbols. More information is in the next section.

## How does it work?
All SVGs icons from phosphor react were inlined on the final bundle because they were server-side. To remove those icons inlined, this PR adds a unique SVG file with all icons as symbols, and reuses them with `<use>` tag. Also, this strategy allow style the SVGs with CSS and inheritance. 

Performance results for all pages:
- 🎉  Remove unused javascript from phosphor-react library.

Performance result at home page:
- Decreased the some performance metrics TBT, LCP, TTI on home (check the lighthouse)
- Decreased the `onInitialClientRender` mark in ~17%

Before bundle:
<img width="1030" alt="image" src="https://user-images.githubusercontent.com/15680320/157344845-29749e64-37f7-44e5-aa72-56a125e82a5b.png">

After bundle:
<img width="1037" alt="image" src="https://user-images.githubusercontent.com/15680320/157344685-f6f9db10-417a-432f-aec8-ac08810a5e11.png">

For more info
https://base.vtex.app/bundle-analyser.html
https://preview-378--base.preview.vtex.app/bundle-analyser.html

https://pagespeed.web.dev/report?url=https%3A%2F%2Fpreview-378--base.preview.vtex.app%2F

## How to test it?
Open the preview, navigate, with fast 3g and without throtling, hover elements.
Check if the element colors, size, widths still the same.

## References
After this approach https://github.com/vtex-sites/base.store/pull/358 i'm trying this one.
https://cloudfour.com/thinks/svg-icon-stress-test/#image-element-with-data-uri

## Checklist

<em>You may erase this after checking them all ;)</em>
